staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -156,6 +158,11 @@ type crdInfo struct {
 	storageVersion string
 
 	waitGroup *utilwaitgroup.SafeWaitGroup
+
+	allowMutation atomic.Bool
+
+	storageVersionUpdateLock sync.Mutex
+	deleted                  bool
```

Comment:
these are used to ensure the REST update is not in progress when the old instance is deleted.